### ISSUE TITLE
Question: Is kafka-go production ready ?

### ISSUE THREAD

**Issue by enghardik** [segmentio/kafka-go#676]
I am planning to use kafka-go for building a production ready Kafka producer client. kafka-go's [Conn](https://pkg.go.dev/github.com/segmentio/kafka-go#Conn) type provides us the exact level of abstraction and set of features required by our use-case.

There are a few factors that are stopping me from using kafka-go over [Shopify's sarama](https://github.com/Shopify/sarama).
1. As per the [Go doc](https://pkg.go.dev/github.com/segmentio/kafka-go), kafka-go is still not stable because its first major release (v1.0) is pending. 
2. Contrary to sarama, kafka-go has not mentioned the organisations using it in production. Is it used in production by any company ? Is it industry accepted ?

Please help me by pointing out some clarifications related to above issues and facts stating that kafka-go is production ready. Also, what's the plan for future releases and when can we expect a v1.0 release ?

**Comment by achille-roussel** [comment]
Hello @enghardik, thanks for opening the issue.

We have been using kafka-go at Segment since we started the project in 2017, and is now the underlying client for most of our large scale production workloads.

We don't advertise a list of companies using kafka-go because we haven't collected it, but the project is actively maintained and we frequently receive external contributions. I believe it has gained popularity over the years.

Regarding v1.0, our intention is not to indicate that the project is not production-ready, but that we may still need to improve some of the APIs, which may break backward compatibility. That being said, we have introduced very few breaking changes since the start of the project, and are committed to avoiding them as much as possible. What will likely happen in v1.0 is we will remove APIs that we have deprecated over time.

I hope this answer was useful, let me know if you have other questions.
